client/ratings_reviews/components/RatingBreakdown.jsx

```diff
@@ -36,30 +36,16 @@ class RatingBreakdown extends React.Component {
             <span class='review-average-rating-number'>{this.state.averageRating[0]}</span>
             <span class="stars" style={{'--rating': this.state.averageRating[1]}}></span>
           </div>
-          <div class='review-recommended-percetage'>
+          <div class='review-recommended-percentage'>
             <span>{this.state.recommended} of reviews recommended this product</span>
           </div>
+          <div class='review-filter-panel'>
+            <div>Rating Breakdown</div>
+            <div>Currently applied filters: {this.props.starFilters.map(starFilter => starFilter + ' ')}</div>
+            <div class='review-clickable' onClick={this.props.removeFilters}>Remove all filters</div>
+          </div>
           <div class='review-rating-bar'>
-            <div class='review-breakdown-bar'>
-              <span>5 stars:</span>
-              <span><progress max="100" value={this.state.breakdown[4]}></progress></span>
-            </div>
-            <div class='review-breakdown-bar'>
-              <span>4 stars:</span>
-              <span><progress max="100" value={this.state.breakdown[3]}></progress></span>
-            </div>
-            <div class='review-breakdown-bar'>
-              <span>3 stars:</span>
-              <span><progress max="100" value={this.state.breakdown[2]}></progress></span>
-            </div>
-            <div class='review-breakdown-bar'>
-              <span>2 stars:</span>
-              <span><progress max="100" value={this.state.breakdown[1]}></progress></span>
-            </div>
-            <div class='review-breakdown-bar'>
-              <span>1 star:</span>
-              <span><progress max="100" value={this.state.breakdown[0]}></progress></span>
-            </div>
+            {[...Array(5).keys()].reverse().map(x => <RatingBreakdownBar star={x + 1} percentage={this.state.breakdown[x][0]} freq={this.state.breakdown[x][1]} handleRatingBreakdownClick={this.handleRatingBreakdownClick}/>)}
```

Comment:
Nice! If I'm reading correctly this line does what the previous 20ish lines of code does?

Reply:
Hi @jabrake, yes I saw some repetition here for the rating breakdown bars, so I created a sub-component for it. This line just creates a temporary array for the breakdown bar component to get rendered 5 times, one for each star. Thank you for asking!